staging/src/k8s.io/apiserver/pkg/endpoints/filters/cachecontrol.go

```diff
@@ -0,0 +1,35 @@
+package filters
+
+import (
+	"net/http"
+	"strings"
+)
+
+// noCachePathPrefixes allows caching for large, cacheable, public endpoints with etags, like openapi
+// see https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching#defining_optimal_cache-control_policy
+var noCachePathPrefixes = []string{
+	"/openapi/",
+}
+
+// WithCacheControl sets the Cache-Control header to no-cache, if it is for openapi and private otherwise because
+// all servers are protected by authn/authz.
+func WithCacheControl(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		// if the cache-control header is already set, let it be.
+		if _, ok := w.Header()["Cache-Control"]; ok {
+			handler.ServeHTTP(w, req)
+			return
+		}
+
+		for _, prefix := range noCachePathPrefixes {
+			if strings.HasPrefix(req.URL.Path, prefix) {
+				w.Header().Set("Cache-Control", "no-cache")
+				handler.ServeHTTP(w, req)
+				return
+			}
+		}
+
+		w.Header().Set("Cache-Control", "private")
```

Comment:
`no-cache, private`

Reply:
I actually think `no-cache, private` is reasonable for all requests... that just makes clients verify the etag before using cached content, which they are doing with openapi content already